source/extensions/transport_sockets/tls/ssl_socket.h

```diff
@@ -41,13 +41,8 @@ struct SslSocketFactoryStats {
 enum class InitialState { Client, Server };
 enum class SocketState { PreHandshake, HandshakeInProgress, HandshakeComplete, ShutdownSent };
 
-class SslSocket : public Network::TransportSocket,
-                  public Envoy::Ssl::ConnectionInfo,
-                  public Envoy::Ssl::PrivateKeyConnectionCallbacks,
-                  protected Logger::Loggable<Logger::Id::connection> {
-public:
-  SslSocket(Envoy::Ssl::ContextSharedPtr ctx, InitialState state,
-            const Network::TransportSocketOptionsSharedPtr& transport_socket_options);
+struct SslSocketInfo : public Envoy::Ssl::ConnectionInfo {
```

Comment:
keep this class and fields as private.

Reply:
it's used directly in ssl_socket_test due to raw SSL access. do you want to add a friend declaration? it doesn't seem to be common to add test friends here.

Reply:
do you need the mutable cached_* fields for tests? it's fine to keep rawSslForTest public.

Reply:
it's doing both dynamic cast and raw SSL access... there's no way to access SslSocket from StreamInfo so it goes through SslSocketInfo.

Reply:
Ugh my first comment wasn't clear. I meant make this `class SslCocketInfo`, and fields in this class after `ssl_` private.

Reply:
done